src/bin/genemichaels.rs

```diff
@@ -201,7 +205,42 @@ fn main() {
         (&src).lines().take(5).any(|l| l.contains("`nogenemichaels`"))
     }
 
-    if args.files.is_empty() {
+    if args.workspace {
+        let mut cargo_toml = None;
+
+        // find workspace cargo toml
+        for parent in 0.. {
+            let dir = if parent == 0 {
+                ".".into()
+            } else {
+                "../".repeat(parent)
+            };
+            let maybe_cargo_toml = std::fs::read_dir(dir)?.filter_map(|f| f.ok()).find(|f| {
+                f.file_type().ok() == Some("toml") && f.file_name().ok() == Some("Cargo")
+            });
+            match maybe_cargo_toml {
+                Some(ct) => {
+                    let string_toml = std::fs::read_to_string(ct);
```

Comment:
Thanks for working on this! Quick thought, I wonder if it would good to use something like https://github.com/LukeMathWalker/cargo-manifest ? (I've used that before, and it worked)

It might be a bit more type safe

Reply:
Interesting - I didn't know there was something that did this properly (I've done something similar myself in the past). Will use!